events.lua

```diff
@@ -59,6 +83,19 @@ function register_events(event)
 				end
 			end
 		end
+
+                gauge_research_queue:reset()
```

Comment:
since you've created research.lua put everything in a func called on_research_tick() in research.lua

Reply:
This is necessary to also get the current research state in terms of progress.
Not only the queue.

This will return the current state:
```
research_queue{index="1"} 
```

Here you get the queued research
```
research_queue{index!="1"} 
```

Reply:
oh. I think I know what you mean.

Reply:
this comment is for line 79 to line 90, it's just to avoid having the code directly in the event loop, like for on_power_tick() for example

Reply:
here you go.
[62f7dbf](https://github.com/remijouannet/graftorio2/pull/24/commits/62f7dbffc43715635fdd1d35166282fe49d01557)

Reply:
I remove `if event.tick` from on_research_finished...